### PR TITLE
feat: Rewards, missed rewards, penalties

### PR DESCRIPTION
**Pull request by vgorkavenko**
This PR proposes many improvements and implementation of rewards, penalties and missed rewards calculation.
 
Here are some of the major:
- Calculate each validator's rewards, penalties, and missed rewards in each epoch (without 'slashing' and 'inactivity leak' cases yet). Add new Grafana Dashboard
- Use the official Clickhouse client and streams for writing the data. Now we don't need `DB_INSERT_CHUNK_SIZE`
- Fix bug with failed high reward validators task
- Fix event-loop lags

### CODE REVIEW

**Review by dgusakov — APPROVED** [review]

**Review by madlabman — APPROVED** [review]
Nice job! I'm unable to actually delve into the rewards mechanic at the moment, LGTM!

File: src/duty/attestation/attestation.service.ts [hunk 1]
```diff
@@ -38,44 +44,62 @@ export class AttestationService {
...
+    ]);
+    const att_valid_head = attestation.head == canonHead;
+    const att_valid_target = attestation.target_root == canonTarget;
+    const att_valid_source = attestation.source_root == canonSource;
+    const att_inc_delay = Number(attestation.included_in_block - attestation.slot);
+    const reward_per_increment = attestationRewards(att_inc_delay, att_valid_source, att_valid_target, att_valid_head);
+    const penalty_per_increment = attestationPenalties(att_inc_delay, att_valid_source, att_valid_target, att_valid_head);
+    for (const [valCommIndex, validatorIndex] of committee.entries()) {
+      const isProcessed = this.summary.get(validatorIndex)?.att_happened;
+      if (isProcessed) continue; // already processed validator. it was in one of previous attestation
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply src/duty/attestation/attestation.service.ts]: done

File: src/duty/duty.service.ts [hunk 2]
```diff
@@ -29,27 +34,41 @@ export class DutyService {
...
-  public async checkAndWrite(epoch: bigint, stateSlot: bigint): Promise<any> {
+  public async checkAndWrite(epoch: bigint, stateSlot: bigint): Promise<string[]> {
     // Prefetch will be done before main checks because duty by state requests are heavy
     // and while we wait for their responses we fetch blocks and headers.
     // If for some reason prefetch task will be slower than duty by state requests,
     // blocks and headers will be fetched inside tasks of checks
-    await Promise.all([this.prefetch(epoch), this.checkAll(epoch, stateSlot)]);
-    await this.write();
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [a, b, possibleHighRewardVals] = await Promise.all([
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply src/duty/duty.service.ts]: done

File: src/duty/summary/summary.metrics.ts [hunk 3]
```diff
@@ -3,19 +3,88 @@ import { Inject, Injectable, LoggerService } from '@nestjs/common';
 
 import { ConfigService } from 'common/config';
 import { ConsensusProviderService } from 'common/eth-providers';
-import { PrometheusService } from 'common/prometheus';
+import { PrometheusService, TrackTask } from 'common/prometheus';
+
+import { RegistryService, RegistrySourceOperator } from '../../common/validators-registry';
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply src/duty/summary/summary.metrics.ts]: done

File: src/duty/attestation/attestation.service.ts [hunk 4]
```diff
@@ -38,44 +44,62 @@ export class AttestationService {
...
-          if (this.summary.get(validatorIndex)?.att_happened) continue;
-          const att_happened = attestation.bits[valCommIndex];
-          let summary = { epoch, val_id: validatorIndex, att_happened };
-          const properties = { att_inc_delay, att_valid_head, att_valid_source, att_valid_target };
-          if (att_happened) summary = { ...summary, ...properties };
-          this.summary.set(validatorIndex, summary);
-        }
+    for (const attestation of attestations) {
+      // Each attestation corresponds to committee. Committee may have several aggregate attestations
+      const committee = committees.get(`${attestation.committee_index}_${attestation.slot}`);
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply src/duty/attestation/attestation.service.ts:55]: `committees` is just hash map with such keys. I suggest to leave this

**Pull request merged**